### DISCUSSION TITLE
Serverless Function has crashed,but no explicit error message in the error logs

### DISCUSSION THREAD

**Discussion by oluwatobiisaiah** [vercel/community#3178] [Help] [answered]
### Summary

I hosted a backend app built with express on vercel, it was working all well before but I just noticed it stopped working and it's giving a 500 internal server error, and on the home directory I am getting "Serverless Function Crashed" and when I check the error logs, there is no explicit error message displayed and everything works fine before on vercel and currently on my localhost.
I don't know if something was updated that caused the sudden crash..
But here is my vercel.json file
```json
{
    "version": 2,
    "builds": [
      {
        "src": "app.js",
        "use": "@now/node"
      }
    ],
    "routes": [
      {
        "src": "/(.*)",
       "dest": "app.js"
      }
    ]
  }
  
```

### Example

https://hiracbt-backend.vercel.app/

### Steps to Reproduce

The error has made all the application crashed, so you can make any request on :https://hiracbt-backend.vercel.app to replicate the problem

**Comment by amyegan** [comment, marked as answer]
Hi @oluwatobiisaiah. In situations like this one, I usually add a [try/catch block](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch) around sections of the code to intentionally print more details. The guide about using Express with Vercel may also help: https://vercel.com/guides/using-express-with-vercel

I would also recommend changing your vercel.json to update the legacy configuration from `builds` to [`functions`](https://vercel.com/docs/concepts/projects/project-configuration#functions) and [from `routes`](https://vercel.com/docs/concepts/projects/project-configuration#upgrading-legacy-routes) to  the newer [`rewrites`](https://vercel.com/docs/concepts/projects/project-configuration#project/rewrites) or [`redirects`](https://vercel.com/docs/concepts/projects/project-configuration#project/redirects) properties. And update to `@vercel/node` instead of `@now/node`.
